### ISSUE TITLE
Pages and Layouts Typescript example broken

### ISSUE THREAD

**Issue by slavanga** [vercel/next.js#49610]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.4.0: Mon Mar  6 21:00:17 PST 2023; root:xnu-8796.101.5~3/RELEASE_X86_64
    Binaries:
      Node: 18.13.0
      npm: 8.19.3
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 13.4.2-canary.4
      eslint-config-next: 13.4.1
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.0.4
```

### Which example does this report relate to?

layout-component

### Describe the Bug
Error on the following line:
```js
return getLayout(<Component {...pageProps} />);
```

```
'Component' cannot be used as a JSX component.
  Its element type 'ReactElement<any, any> | Component<any, any, any> | null' is not a valid JSX element.
    Type 'Component<any, any, any>' is not assignable to type 'Element | ElementClass | null'.
      Type 'Component<any, any, any>' is not assignable to type 'ElementClass'.
        The types returned by 'render()' are incompatible between these types.
          Type 'React.ReactNode' is not assignable to type 'import("/Users/…/Documents/Projects/…/node_modules/@types/react-dom/node_modules/@types/react/index").ReactNode'.
            Type 'ReactElement<any, string | JSXElementConstructor<any>>' is not assignable to type 'ReactNode'.
              Property 'children' is missing in type 'ReactElement<any, string | JSXElementConstructor<any>>' but required in type 'ReactPortal'.ts(2786)
```

### Expected Behavior

no typescript errors

### To Reproduce

check example for typescript errors

**Comment by stephenasuncionDEV** [comment]
I am also getting this when using dynamic imports.

```ts
const General = dynamic(() => import("@/components/Account/General"), {
  loading: () => <Loading />,
});
```
```ts
  <AccountLayout curPage={curPage}>
        {
          {
            general: <General />,
            ...
          }[curPage]
        }
      </AccountLayout>
```
Error:
General cannot be used as a JSX component.

**Comment by Altroo** [comment]
Adding this to your package.json solves it temporary until a fix is made.

`  "resolutions": {
    "@types/react": "^17.0.38"
  }`

**Comment by freeatnet** [comment]
https://github.com/vercel/next.js/issues/42292#issuecomment-1555975533

**Comment by slavanga** [comment]
@freeatnet thanks for cross posting. re-installing `@types/react-dom` worked for me.

**Comment by stephenasuncionDEV** [comment]
> [#42292 (comment)](https://github.com/vercel/next.js/issues/42292#issuecomment-1555975533)
> 
> > I faced a similar issue after upgrading `@types/react` from `18.2.0` to `18.2.6` alongside `@types/react-dom` from `18.2.1` to `18.2.4`.
> > TL;DR: Try deduplicating `@types/react` in your lockfile, or remove and re-install `@types/react-dom`.
> > The underlying issue appeared to be that `@types/react-dom` lists `@types/react@*` as a dependency, so both `@types/react@18.2.6` and `@types/react-dom > @types/react@18.2.0` ended up being installed after the upgrade. This results in some sort of a type conflict and produces the error above.

Unfortunately does not work in my case.

**Comment by freeatnet** [comment]
@stephenasuncionDEV if you would like some help, could you post a minimal reproduction of the issue?

**Comment by PaulRBerg** [comment]
Did some debugging on this, and it appears that the bug happens with all `@types/react` versions from `18.2.4` all the way to `18.2.8` (inclusive). Downgrading to `18.2.3` (or earlier) fixes the bug.

Also, FYI, it only occurs when `next/dynamic` components:

```typescript
import dynamic from "next/dynamic";

const Shimmer = dynamic(() => import("./component"), {
  ssr: false,
});

export default Shimmer;
```

**Comment by Raduc4** [comment]
Has anyone solved this error on Arch Linux?
`ERROR: Could not find dependency: libicui18n.so.72`
```
pacman -Qi icu
Name            : icu
Version         : 73.2-2
```
